### PR TITLE
HuggingFace Spaces server deployment option

### PR DESCRIPTION
**Pull request by strickvl**
Added a `Dockerfile` to be used by default in Huggingface Spaces to deploy the ZenML server. Also adds documentation for such a deployment.

A photo and a button will need to be added after we're made 'official' on Huggingface.

This PR includes a draft of the documentation that would be included on Huggingface's own native docs -- i.e. like Argilla have [here](https://huggingface.co/docs/hub/spaces-sdks-docker-argilla). Parts of it are replicated in the docs page created on this PR). Once approved, I'll submit the `.md` file and the image asset in a PR on [Huggingface's docs repo](https://github.com/huggingface/hub-docs/tree/main/docs/hub).

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docker/zenml-server-hf-spaces.Dockerfile
Comment (schustmi) [inline docker/zenml-server-hf-spaces.Dockerfile:2]: I'm wondering, can we simply replace this by something like:

```
ARG PYTHON_VERSION=3.9
ARG ZENML_VERSION

FROM zenmldocker/zenml-server:{ZENML_VERSION:+==$ZENML_VERSION}-py{$PYTHON_VERSION}
```

to save duplicating the entire file?

**Review by schustmi — CHANGES_REQUESTED** [review]

File: docker/zenml-server-hf-spaces.Dockerfile
Comment (strickvl) [reply docker/zenml-server-hf-spaces.Dockerfile:2]: ok let me try that. Most of the other spaces don't use an `ENTRYPOINT` whereas we do, so I wanted to allow people to pass in their custom ENV variables.

Comment (strickvl) [reply docker/zenml-server-hf-spaces.Dockerfile]: I updated it as sparsely as seemed to work. It still requires overwriting the `ENTRYPOINT` and `CMD` instructions, but indeed much better this way.

File: docker/zenml-server-hf-spaces.Dockerfile
Comment (strickvl) [inline docker/zenml-server-hf-spaces.Dockerfile:4]: This ENV variable is not being picked up by Mixpanel. Not sure if there's a better way to handle making sure analytics are turned on for this deployment type. @bcdurak for when you're back?

File: docs/book/drafts/spaces-sdks-docker-zenml.md
Comment (htahir1) [inline docs/book/drafts/spaces-sdks-docker-zenml.md:47]: Which photo :-)

File: docs/book/getting-started/deploying-zenml/huggingface.md
Comment (htahir1) [inline docs/book/getting-started/deploying-zenml/huggingface.md:28]: Which button :-)

**Review by htahir1 — CHANGES_REQUESTED** [review]

**Comment by strickvl** [comment]
@htahir1 perhaps you want to give this a glance over. This includes a text (the one inside the `drafts` folder) that will go up on the Huggingface docs site, so possible first point of call for people ever learning about ZenML...

File: docs/book/drafts/spaces-sdks-docker-zenml.md
Comment (strickvl) [reply docs/book/drafts/spaces-sdks-docker-zenml.md:47]: We need our space to become an official space first, so we can take a photo that shows it as a clickable option. They said this would be no problem, but we need that to happen first.

Comment (htahir1) [reply docs/book/drafts/spaces-sdks-docker-zenml.md:47]: So should we take it out until that happens?

File: docs/book/drafts/spaces-sdks-docker-zenml.md
Comment (htahir1) [inline docs/book/drafts/spaces-sdks-docker-zenml.md:90]: is the --no-verify-ssl neccessary? What about the password?

File: docs/book/drafts/spaces-sdks-docker-zenml.md
Comment (htahir1) [inline docs/book/drafts/spaces-sdks-docker-zenml.md:142]: Id suggest people to go to the docs starter guide here instead as we might deprecate ZenBytes in the near future in favor of docs (Something in the works)

File: docs/book/drafts/spaces-sdks-docker-zenml.md
Comment (htahir1) [inline docs/book/drafts/spaces-sdks-docker-zenml.md:139]: Maybe another step here would be to "invite" a user and make them admin to change the default username and password and then reconnect with `zenml connect`

**Review by htahir1 — CHANGES_REQUESTED** [review]
Looks good, had minor comments

File: docs/book/drafts/spaces-sdks-docker-zenml.md
Comment (strickvl) [reply docs/book/drafts/spaces-sdks-docker-zenml.md:47]: I think their space will be created before this gets merged tbh. Once I get the go-ahead from @schustmi on the `Dockerfile` I'll make a PR and let them know we'd like them to set us up as official. It'll be handled.

**Review by schustmi — APPROVED** [review]

**Comment by strickvl** [comment]
HF docs PR made [here](https://github.com/huggingface/hub-docs/pull/717) just now. Will wait for the official space (and the photo) before merging on our side.

File: docker/zenml-server-hf-spaces.Dockerfile
Comment (stefannica) [inline docker/zenml-server-hf-spaces.Dockerfile:1]: I'm a bit puzzled about what this file is for and how it relates to the docs.

File: docs/book/drafts/spaces-sdks-docker-zenml.md
Comment (strickvl) [reply docs/book/drafts/spaces-sdks-docker-zenml.md:1]: The description of the PR has the context. This is a draft for the HF spaces docs, which I used to submit a PR.

File: docker/zenml-server-hf-spaces.Dockerfile
Comment (strickvl) [reply docker/zenml-server-hf-spaces.Dockerfile:1]: This is the Dockerfile used to spin up our official template on the Hub / in Spaces. You can see it in use [here](https://huggingface.co/spaces/zenml/zenml/tree/main).

**Comment by stefannica** [comment]
All good on my end.

**Pull request merged**
